Add info on default resources (#205)
rancher/fleet-docs: docs/ref-gitrepo.md

  # type. Secrets must be of the type "kubernetes.io/ssh-auth" or
  # "kubernetes.io/basic-auth". The secret is assumed to be in the
  # same namespace as the GitRepo

>   # Fleet always checks for a secret named "gitcredential".

Reviewer: ```suggestion
  # If no clientSecretName is supplied, Fleet checks for a secret named
  # "gitcredential".
```

>   # this service account already exists so it should be create before
>   # hand, most likely coming from another git repo registered with

Reviewer: ```suggestion
  # this service account already exists so it should be created beforehand,
  # for instance coming from another git repo registered with
```

  # the Fleet manager.

>   # Fleet always checks for a service account named "fleet-default".

Reviewer: ```suggestion
  # If no service account is configured, Fleet checks for a service account
  # named "fleet-default".
```
